packages/webdriverio/src/commands/browser/uploadFile.ts

```diff
@@ -41,20 +41,15 @@ export default async function uploadFile (localPath) {
         throw new Error(`The uploadFile command is not available in ${this.capabilities.browserName}`)
     }
 
-    let zipData = []
+    let zipData: Uint8Array[] = []
     let source = fs.createReadStream(localPath)
 
     return new Promise((resolve, reject) => {
         archiver('zip')
-            .on('error', (err) => reject(err))
-            .on('data', (data) => zipData.push(data))
+            .on('error', (err: Error) => reject(err))
+            .on('data', (data: Uint8Array) => zipData.push(data))
             .on('end', () => this.file(Buffer.concat(zipData).toString('base64')).then(resolve, reject))
             .append(source, { name: path.basename(localPath) })
-            .finalize((err) => {
-                /* istanbul ignore next */
-                if (err) {
```

Comment:
is it ok that errors will be ignored?

Reply:
yes, TS has shown that `finalize` doesn't take callbacks anymore, so this is a overdue clean up